op-chain-ops/safe/batch_file.go

```diff
@@ -0,0 +1,132 @@
+// Package safe contains types for working with Safe smart contract wallets. These are used to
+// build batch transactions for the tx-builder app. The types are based on
+// https://github.com/safe-global/safe-react-apps/blob/development/apps/tx-builder/src/typings/models.ts.
+package safe
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// BatchFile represents a Safe tx-builder transaction.
+type BatchFile struct {
+	Version      string             `json:"version"`
+	ChainID      *big.Int           `json:"chainId"`
+	CreatedAt    uint64             `json:"createdAt"`
+	Meta         BatchFileMeta      `json:"meta"`
+	Transactions []BatchTransaction `json:"transactions"`
+}
+
+// bathcFileMarshaling is a helper type used for JSON marshaling.
+type batchFileMarshaling struct {
```

Comment:
why do you need the `Marshaling` types? they appear to be exactly the same as the primary type.

Reply:
The types are not exactly the same, if you see the string vs big.int. We could get rid of the types and use anon structs in the marshal methods but opted to not do that, geth uses this pattern but they autogen the marshal types

Reply:
Why does marshalling the big int not work natively?

Reply:
heh, so you do need a different type to marshall it - could be custom BigInt or just do the string. Make sense.